core/common.ts

```diff
@@ -320,21 +320,26 @@ export function defineBlocks(blocks: {[key: string]: BlockDefinition}) {
  * @param e Key down event.
  */
 export function globalShortcutHandler(e: KeyboardEvent) {
-  const mainWorkspace = getMainWorkspace() as WorkspaceSvg;
-  if (!mainWorkspace) {
-    return;
+  let workspace: WorkspaceSvg = getMainWorkspace() as WorkspaceSvg;
+  const focusedTree = getFocusManager().getFocusedTree();
+  for (const ws of getAllWorkspaces()) {
+    if (focusedTree === (ws as WorkspaceSvg)) {
+      workspace = ws as WorkspaceSvg;
+      break;
+    }
   }
 
   if (
     browserEvents.isTargetInput(e) ||
-    (mainWorkspace.rendered && !mainWorkspace.isVisible())
+    !workspace ||
+    (workspace.rendered && !workspace.isFlyout && !workspace.isVisible())
```

Comment:
why did you have to add the `isFlyout` check here? wouldn't we want to not do keyboard shortcuts on a not-visible flyout workspace too?

Reply:
`isVisible()` on a flyout workspace is a lie. `Flyout.setVisible()` [doesn't update the visibility of its workspace](https://github.com/google/blockly/blob/develop/core/flyout_base.ts#L490-L502), and changing it to do so breaks other things, although I forget what specifically, but it [vaguely warns against it](https://github.com/google/blockly/blob/develop/core/workspace_svg.ts#L1266). It felt like enough of a Chesterton's Fence that I decided to punt on it (and also, if the workspace is focused, it must be visible anyway).